compilation/src/main/scala/scala/tools/nsc/HotSbtBenchmark.scala

```diff
@@ -15,80 +15,60 @@ import org.openjdk.jmh.annotations._
-
-  @Param(value = Array("0.13.15"))
-  var sbtVersion: String = _
+  @Param(value = Array())
+  var projectName: String = _
 
   @Param(value = Array(""))
-  var extraBuild: String = _
-
-  @Param(value= Array("compile"))
-  var compileCmd: String = _
-
-  // This parameter is set by ScalacBenchmarkRunner / UploadingRunner based on the Scala version.
-  // When running the benchmark directly the "latest" symlink is used.
-  @Param(value = Array("latest"))
-  var corpusVersion: String = _
+  var extraArgs: String = _
 
   var sbtProcess: Process = _
   var inputRedirect: ProcessBuilder.Redirect = _
   var outputRedirect: ProcessBuilder.Redirect = _
-  var tempDir: Path = _
-  var scalaHome: Path = _
+  var path: Path = _
+  var cleanClassesPath: Path = _
   var processOutputReader: BufferedReader = _
   var processInputReader: BufferedWriter = _
   var output= new java.lang.StringBuilder()
 
-  def compileRawSetting = if (sbtVersion.startsWith("1.")) "" else
-    """
-      |val compileRaw = taskKey[Unit]("Compile directly, bypassing the incremental compiler")
-      |
-      |def addCompileRaw = compileRaw := {
-      |    val compiler = new sbt.compiler.RawCompiler(scalaInstance.value, sbt.ClasspathOptions.auto, streams.value.log)
-      |    classDirectory.value.mkdirs()
-      |    compiler.apply(sources.value, classDirectory.value +: dependencyClasspath.value.map(_.data), classDirectory.value, scalacOptions.value)
-      |  }
-      |
-      |inConfig(Compile)(List(addCompileRaw))
-    """.stripMargin
-
-  def buildDef =
-    s"""
-       |scalaHome := Some(file("${scalaHome.toAbsolutePath.toString}"))
-       |
-       |val cleanClasses = taskKey[Unit]("clean the classes directory")
+  private def cleanClassesPlugin: String =
+    s"""package bloop
+       |import sbt._
+       |import Keys._
        |
-       |cleanClasses := IO.delete((classDirectory in Compile).value)
+       |object CleanClassesPlugin extends AutoPlugin {
+       |  override def trigger = allRequirements
+       |  override def requires = plugins.JvmPlugin
        |
-       |scalaSource in Compile := file("${corpusSourcePath.toAbsolutePath.toString}")
+       |  object autoImport {
+       |    lazy val cleanClasses = taskKey[Unit]("Clean all classes in all projects")
+       |    lazy val cleanClassesTask = taskKey[Unit]("Clean all classes")
+       |  }
        |
-       |libraryDependencies += "org.scala-lang" % "scala-compiler" % scalaVersion.value
-       |libraryDependencies += "org.scala-lang" % "scala-reflect" % scalaVersion.value
+       |  import autoImport._
        |
-       |traceLevel := Int.MaxValue
-       |traceLevel in runMain := Int.MaxValue
+       |  override def globalSettings: Seq[Setting[_]] = Seq(
+       |    cleanClasses := Def.taskDyn {
+       |      cleanClassesTask.all(ScopeFilter(inAnyProject))
+       |    }.value
+       |  )
        |
-       |$compileRawSetting
-       |
-       |$extraBuild
-       |// TODO support .java sources
```

Comment:
Do we support Java sources?

Reply:
As long as we don't run the `ScalacBenchmark`, yes.